xml_converter/generators/code_generator.py

```diff
@@ -286,8 +295,10 @@ class AttributeVariable:
     xml_fields: List[str]
     protobuf_field: str
     default_xml_fields: List[str] = field(default_factory=list)
-    xml_export: str = ""
-    is_child: bool = False
+    xml_child_export: List[str] = field(default_factory=list)
+    xml_parent_export: str = ""
+    compound_name: Optional[str] = None
```

Comment:
I see why this variable is necessary, however it is both confusingly named and confusingly used in the template.

In effect it is required for when we want to do

```
this->color.red = [some parser]
```
instead of
```
this->red = [some parser
```

However some of the templating logic surrounding it is strange
https://github.com/AsherGlick/Burrito/pull/134/files?diff=split&w=0#diff-52828a1048f08c49e1346b0bd8a90d4c4210421c8c5bf9546108765497e30b85R500

It is too difficult to piece out the templating changes due to the weird merge diffs so I will revisit this part of the review after the other PRs are ready to be merged.